lnwallet/size.go

```diff
@@ -301,3 +327,65 @@ func estimateCommitTxWeight(count int, prediction bool) int64 {
 
 	return htlcWeight + baseWeight + witnessWeight
 }
+
+// TransactionWeightEstimate is able to calculate weight estimates for
+// transactions based on the input and output types. For purposes of estimation,
+// all signatures are assumed to be of the maximum possible size, 73 bytes.
+type TransactionWeightEstimate struct {
```

Comment:
Love this addition, this'll prove to be very useful within the codebase!

Reply:
Also a set of equivalent methods for adding the weight of nested `p2sh` outputs should be added. For simplicity, for now we can assume that if the `pkScript` for an input is p2sh-like, then it's spending a nested p2sh output. 

To accomplish this, we can extend the `lnwallet.Utxo` struct to also include the `pkScript` of the created output. Better perhaps, we can use a new num to denote the type of the script. There's already values like `lnwallet.NestedWitnessPubkey` defined, so we can piggy back off of those.

Reply:
Naming suggestion: `TxWeightEstimator`